File: framework-ts/module/devian-domain-game/features/ability/AbilityCard.ts
import { ITEM_CARD } from '../../Generated/Game.g';
import { AbilityBase } from './AbilityBase';

export class AbilityCard extends AbilityBase {
    private mTable: ITEM_CARD | null = null;

    get cardId(): string { return this.mTable?.CardId ?? ''; }

    init(table: ITEM_CARD): void {
        this.mTable = table;
    }

    clone(): AbilityCard {
        const c = new AbilityCard();
        c.mTable = this.mTable;
        c.copyStatsFrom(this);
        return c;
    }
}

File: framework-cs/upm/com.devian.domain.game/Runtime/Ability/AbilityCard.cs
using Devian.Domain.Game;

namespace Devian
{
    public sealed class AbilityCard : AbilityBase
    {
        ITEM_CARD mTable = null;

        public string CardId => mTable?.CardId ?? string.Empty;
        public int Amount => this[STAT_TYPE.CARD_AMOUNT];

        public void Init(ITEM_CARD table)
        {
            mTable = table;
        }

        public override AbilityBase Clone()
        {
            var c = new AbilityCard();
            c.mTable = mTable;
            c.CopyStatsFrom(this);
            return c;
        }

        public void AddAmount(int delta)
        {
            AddStat(STAT_TYPE.CARD_AMOUNT, delta);
        }
    }
}
